py3status/modules/rss_aggregator.py

```diff
@@ -0,0 +1,132 @@
+given feed ID and a given folder ID will be counted only once.
+
+Configuration parameters:
+    aggregator: feed aggregator used, see note below. For now, the only supported value is
+        `'owncloud'`, that means that you'll only be able to configure a
+        OwnCloud/NextCloud instance. Other feed aggregator may be supported in future
+        releases. (default 'owncloud')
+    cache_timeout: how often to run this check (default 60)
+    feed_ids: list of IDs of feeds to watch, see note below (default [])
+    folder_ids: list of IDs of folders ro watch (default [])
+    format: format to display (default 'Feed: {unseen}')
+    password: login password (default None)
+    server: aggregator server to connect to (default 'https://yourcloudinstance.com')
+    user: login user (default None)
+
+Format placeholders:
+    {unseen} sum of numbers of unread feed elements
+
+Color options:
+    color_new_items: text color when there is new items (default color_good)
+    color_error: text color when there is an error (default color_bad)
+
+Requires:
+    requests: python module from pypi https://pypi.python.org/pypi/requests
+
+@author raspbeguy
+"""
+
+import requests
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    aggregator = 'owncloud'
+    cache_timeout = 60
+    feed_ids = []
+    folder_ids = []
+    format = 'Feed: {unseen}'
+    password = None
+    server = 'https://yourcloudinstance.com'
+    user = None
+
+    def post_config_hook(self):
+        if self.aggregator not in ['owncloud']:  # more options coming
+            raise ValueError('%s is not a supported feed aggregator' % self.aggregator)
+        if self.aggregator == "owncloud":
+            self._verify_owncloud()
+        if self.user is None or self.password is None:
+            raise ValueError("user and password must be provided")
+
+    def _verify_owncloud(self):
+        api_url = "%s/index.php/apps/news/api/v1-2/" % self.server
+        try:
+            r = requests.get(api_url + "feeds",
+                             auth=(self.user, self.password),
+                             timeout=10)
+        except requests.exceptions.MissingSchema:
+            raise ValueError('Unconsistent server URL')
+        except (
+            requests.exceptions.ConnectionError,
+            requests.exceptions.ConnectTimeout
+        ):
+            return
+        if r.status_code != 200:
+            raise ValueError('Not sure this is a Owncloud/Nextcloud instance')
+        else:
+            try:
+                r.json()
+            except:
+                raise ValueError('Server returning bad content, check credentials')
+
+    def check_news(self):
+        if self.aggregator == "owncloud":
+            rss_count = self._get_count_owncloud()
+
+        response = {'cached_until': self.py3.time_in(self.cache_timeout)}
+
+        if rss_count is None:
+            response['full_text'] = "Unreachable"
```

Comment:
I'm not sure it would be comfortable for the user to get a big red banner each time internet is unreachable, especially if all the modules using internet have this strategy (IMAP for instance could do it, even if it isn't the case yet)